nixos/modules/security/pam.nix

```
@@ -98,13 +98,27 @@ let
   package = config.security.pam.package;
   parentConfig = config;
 
-  pamOpts = { config, name, ... }: let cfg = config; in let config = parentConfig; in {
+  valueFromPath = path: value:
+    if builtins.length path == 0
+    then value
+    else valueFromPath (lib.tail path) value.${lib.head path};
```

there is `lib.attrsets.attrByPath` that does that.
```suggestion
  valueFromPath = path: value: lib.attrByPath path (throw "unknown path") value;
```

---

oh lovely